OpenEdXMobile/src/main/java/org/edx/mobile/module/analytics/AnalyticsProvider.java

```diff
@@ -0,0 +1,327 @@
+package org.edx.mobile.module.analytics;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import com.google.inject.Singleton;
+
+import org.edx.mobile.util.images.ShareUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * An analytics provider class whose function is to contain all the analytics services we are using
+ * in our app, and call their respective APIs to send out analytics events.
+ */
+@Singleton
+public class AnalyticsProvider implements Analytics {
+    @NonNull
+    private List<Analytics> services = new ArrayList<>();
+
+    public void addAnalyticsProvider(@NonNull Analytics provider) {
+        services.add(provider);
+    }
+
+    public void trackScreenView(String screenName) {
+        trackScreenView(screenName, null, null);
+    }
+
+    public void trackScreenView(@NonNull String screenName, @Nullable String courseId,
+                                @Nullable String action) {
+        trackScreenView(screenName, courseId, action, null);
+    }
+
+    @Override
+    public void trackScreenView(@NonNull String screenName, @Nullable String courseId,
+                                @Nullable String action, @Nullable Map<String, String> values) {
+        for (Analytics service : services) {
+            service.trackScreenView(screenName, courseId, action, values);
+        }
+    }
+
+    @Override
+    public void trackOpenInBrowser(String url) {
+        for (Analytics service : services) {
+            service.trackOpenInBrowser(url);
+        }
+    }
+
+    @Override
+    public void trackOpenInBrowser(String blockId, String courseId, boolean isSupported) {
+        for (Analytics service : services) {
+            service.trackOpenInBrowser(blockId, courseId, isSupported);
+        }
+    }
+
+    @Override
+    public void trackDownloadComplete(String videoId, String courseId, String unitUrl) {
+        for (Analytics service : services) {
+            service.trackDownloadComplete(videoId, courseId, unitUrl);
+        }
+    }
+
+    @Override
+    public void trackUserFindsCourses() {
+        for (Analytics service : services) {
+            service.trackUserFindsCourses();
+        }
+    }
+
+    @Override
+    public void trackUserConnectionSpeed(String connectionType, float connectionSpeed) {
+        for (Analytics service : services) {
+            service.trackUserConnectionSpeed(connectionType, connectionSpeed);
+        }
+    }
+
+    public void certificateShared(String courseId, String certificateURL,
```

Comment:
I think this is a nice utility method, which allows us to not have custom-type params in our analytics functions like `ShareUtils.ShareType`, keeping all the complexities in the Provider class and giving primitive-types data to concrete classes like `String`, `int` etc.

Lemme add `@NonNull` on the params though.